packages/ms-authenticator/src/authenticators/MsAuthenticator.ts

```diff
@@ -0,0 +1,126 @@
+import { ConfidentialClientApplication, LogLevel, PublicClientApplication, UsernamePasswordRequest } from '@azure/msal-node'
+import {
+  IMsAuthenticationAuthorizationCodeArgs,
+  IMsAuthenticationClientCredentialArgs,
+  IMsAuthenticationOnBehalfOfArgs, IMsAuthenticationSilentFlowArgs,
+  IMsAuthenticationUsernamePasswordArgs,
+} from '../index'
+
+import { fetch } from 'cross-fetch'
+
+/**
+ * Not implemented yet
+ * @param authenticationArgs
+ * @constructor
+ */
+export async function AuthorizationCodeAuthenticator(authenticationArgs: IMsAuthenticationAuthorizationCodeArgs): Promise<string> {
+  throw new Error("This authentication method is not implemented yet.")
+}
+
+/**
+ * Not implemented yet
+ * @param authenticationArgs
+ * @constructor
+ */
+export async function BehalfOfAuthenticator(authenticationArgs: IMsAuthenticationOnBehalfOfArgs): Promise<string> {
+  throw new Error("This authentication method is not implemented yet.")
+}
+
+/**
+ * necessary fields are:
+ *   azClientId: clientId of the application you're trying to login
+ *   azClientSecret: secret of the application you're trying to login
+ *   azTenantId: your MS Azure tenantId
+ *   credentialManifest: address of your credential manifest. usually in following format:
+ *    https://beta.eu.did.msidentity.com/v1.0/<tenant_id>/verifiableCredential/contracts/<verifiable_credential_schema>
+ * @param authenticationArgs
+ * @constructor
+ */
+export async function ClientCredentialAuthenticator(authenticationArgs: IMsAuthenticationClientCredentialArgs): Promise<string> {
+  var msalConfig = {
+    auth: {
+      clientId: authenticationArgs.azClientId,
+      authority: 'https://login.microsoftonline.com/' + authenticationArgs.azTenantId,
+      clientSecret: authenticationArgs.azClientSecret,
+    },
+    system: {
+      loggerOptions: {
+        piiLoggingEnabled: false,
+        logLevel: LogLevel.Verbose,
+      }
+    }
+  }
+
+  const cca = new ConfidentialClientApplication(msalConfig)
+  const msalClientCredentialRequest = {
+    scopes: ['3db474b9-6a0c-4840-96ac-1fceb342124f/.default'],
+    skipCache: false,
+  }
+  await fetch('https://login.microsoftonline.com/' + authenticationArgs.azTenantId + '/v2.0/.well-known/openid-configuration', {method: 'GET'})
+  .then((res) => res.json())
```

Comment:
Are we only doing this to double-check the region. If so I am not sure I like the overhead on every invocation. Then I would rather see a param for the region, against which the authArgs are being checked

Reply:
Ping